src/components/flat-table/flat-table.style.ts

```diff
@@ -9,12 +9,16 @@ import { StyledFlatTableCell } from "./flat-table-cell/flat-table-cell.style";
 import cellSizes from "./cell-sizes.style";
 import Box from "../box";
 import { StyledPagerContainer } from "../pager/pager.style";
+import { FlatTableProps } from "./flat-table.component";
+import { DrawerSidebarContextProps } from "../drawer";
 
 const HEADER_OVERLAY_INCREMENT = 3;
 const STICKY_FOOTER_OVERLAY_INCREMENT = 1;
 const ROW_HEADER_OVERLAY_INCREMENT = 5;
 
-const StyledTableContainer = styled.div`
+const StyledTableContainer = styled.div<
+  Pick<FlatTableProps, "width" | "overflowX">
+>`
   display: grid;
```

Comment:
note: lines 22 & 23 will be removed too when you resolve the conflicts with this PR due to the revert/fix I did recently.